CHANGELOG.md

```diff
@@ -4,6 +4,494 @@ All notable changes to Router will be documented in this file.
+
+If you explicitly create a `RouterHttpServer` handle,
+dropping it while the server is running instructs the server shut down gracefuly.
+However with the handle dropped, there is no way to wait for shutdown to end
+or check that it went without error.
+Instead, the new `shutdown` async method can be called explicitly
+to obtain a `Result`:
+
+```diff
+ use RouterHttpServer;
+ let server = RouterHttpServer::builder().schema("schema").start();
+ // …
+-drop(server);
++server.shutdown().await.unwrap(); 
+```
+
+By [@SimonSapin](https://github.com/SimonSapin)
+
+### Added `apollo_router::TestHarness` ([PR #1487](https://github.com/apollographql/router/pull/1487))
+
+This is a builder for the part of an Apollo Router that handles GraphQL requests,
+as a `tower::Service`.
+This allows tests, benchmarks, etc
+to manipulate request and response objects in memory without going over the network.
+See the API documentation for an example. (It can be built with `cargo doc --open`.)
+
+By [@SimonSapin](https://github.com/SimonSapin)
+
+### Remove telemetry configuration hot reloading ([PR #1501](https://github.com/apollographql/router/pull/1501))
+
+The `map_deferred_response` method is now available for the router service and execution
+service in Rhai. When using the `@defer` directive, we get the data in a serie of graphql
+responses. The first one is available with the `map_response` method, where the HTTP headers
+and the response body can be modified. The following responses are available through
+`map_deferred_response`, which only has access to the response body.
+
+By [@geal](https://github.com/geal) in https://github.com/apollographql/router/pull/1501
+
+## 🐛 Fixes
+
+### Variables validation: return a 400 if variables validation fails ([#1403](https://github.com/apollographql/router/issues/1403))
+
+Failure to validate variables against a query and a schema will now return an HTTP 400.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o)
+
+### Expose query plan: move the behavior to the execution_service ([#1541](https://github.com/apollographql/router/issues/1541))
+
+There isn't much use for QueryPlanner plugins. Most of the logic done there can be done in `execution_service`. Moreover users could get inconsistent plugin behavior because it depends on whether the QueryPlanner cache hits or not.
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o)
+
+### Include usage reporting data in the context even when the query plan has been cached ([#1559](https://github.com/apollographql/router/issues/1559))
+
+Include usage reporting data in the context even when the query plan has been cached when calling `CachingQueryPlanner`.
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/1559
+
+### Accept SIGTERM as shutdown signal ([PR #1497](https://github.com/apollographql/router/pull/1497))
+
+This will make containers stop faster as they will not have to wait until a SIGKILL to stop the router.
```

Comment:
```suggestion
### Accept `SIGTERM` as shutdown signal ([PR #1497](https://github.com/apollographql/router/pull/1497))

This will make containers stop faster as they will not have to wait until a `SIGKILL` to stop the router (which generally comes several seconds later).
```